### DISCUSSION TITLE
Starting a tauri app (or window) with a keyboard shortcut

### DISCUSSION THREAD

**Discussion by joepio** [tauri-apps/tauri#5281] [Q&A]
I'm working on a raycast / alfred type of launcher, which users should be able to open with a simple keyboard combination, even if there is no tauri window open.

I tried using the `@tauri-apps/api/globalShortcut` `register` API, but that no longer works when the window is closed. I could keep the window open by calling `hide` on it, but that seems like an ugly solution.

Maybe I'm missing an API (on the Rust side), or maybe I should use a radically different approach.

**Comment by FabianLars** [comment]
If you want to completely close the window instead of hiding you indeed need to use Rust to register it instead, because without a window the js code can't run.
Here's a small example from one of our repo example: https://github.com/tauri-apps/tauri/blob/b02fc90f450ff9e9d8a35ee55dc1beced4957869/examples/api/src-tauri/src/main.rs#L197-L209

**Comment by david-plugge** [comment]
I have the same use case aswell, opening an overlay on a Keyboard shortcut. I got it working but i wonder if it would be better to hide the window instead of closing. It does feel a little more responsive. Is there a best practice?

**Reply by scottyantipa** [reply]
Ive been trying to use tauri v2 with the `tauri-plugin-global-shortcut` plugin but haven't gotten it working completely. I set the `alpha` of the window to 0 to hide it, which works nicely. However the keyboard shortcuts aren't getting registered on the Rust side (only JS which doesnt work if the window isnt focused).
